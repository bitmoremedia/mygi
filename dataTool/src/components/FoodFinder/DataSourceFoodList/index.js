import React, { PropTypes } from 'react'
import ToolTip from '../../../components/common/ToolTip'

const DataSourceFoodList = ({ list }) => {
  const renderItem = food =>
    <li key={food.id}>
      {food.name} [{food.gi}]
      <ToolTip
        text={`${food.source} / ${food.id}`}
        position={'left'}
      >
        ?
      </ToolTip>
    </li>

  return (
    <div>
      <ul>
        {list.map(food => renderItem(food))}
      </ul>
    </div>
  )
}

DataSourceFoodList.propTypes = {
  list: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
      gi: PropTypes.any.isRequired,
      source: PropTypes.string.isRequired,
    }),
  ).isRequired,
}

export default DataSourceFoodList
