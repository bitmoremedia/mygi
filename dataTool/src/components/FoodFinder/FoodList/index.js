import React from 'react';

// import { Container } from './styles';

const foodListProps = {
  list: React.PropTypes.arrayOf(
    React.PropTypes.shape({
      id: React.PropTypes.string.isRequired,
      name: React.PropTypes.string.isRequired,
      gi: React.PropTypes.any.isRequired,
    })
  ).isRequired,
};

const FoodList = ({ list }) => {
  return (
    <div>
      <ul>
        {list.map((food)=><li key={food.id}>{food.name} [{food.gi}]</li>)}
      </ul>
    </div>
  );
};

FoodList.propTypes = foodListProps;

export default FoodList;
