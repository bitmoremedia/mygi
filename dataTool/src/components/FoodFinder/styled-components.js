import styled from 'styled-components'

export const Container = styled.div`
  padding: 5px;
`

export const Heading = styled.h1`
`

export const SubHeading = styled.h4`
`
