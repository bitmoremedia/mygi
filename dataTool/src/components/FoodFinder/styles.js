import styled from 'styled-components';

export const Container = styled.div`
  padding: 5px;
`;

export const Heading = styled.h1`
`;

export const SubHeading = styled.h4`
`;

export const FoodHeading = styled.span`
  padding-left: 16px;
  font-size: 16px;
`;
