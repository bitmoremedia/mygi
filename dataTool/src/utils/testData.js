const testData = {
  foodList: {
    'abcd-123': {
      category: 'Beans',
      gi: 35,
      id: 'abcd-123',
      name: 'Kidney Beans',
      sources: {},
    },
    'efgh-456': {
      category: 'Bread',
      gi: 68,
      id: 'efgh-456',
      name: 'White Bread',
      sources: {
        havard: 'havard-white-bread',
      },
    },
  },
  dataSources: {
    havard: {
      name: 'havard',
      title: 'Havard',
      data: {
        'havard-white-bread': {
          gi: 65,
          id: 'havard-white-bread',
          name: 'White Bread',
        },
      },
    },
    mendosa: {
      name: 'mendosa',
      title: 'Mendosa',
      data: {
        'mendosa-white-bread': {
          gi: 77,
          id: 'mendosa-white-bread',
          name: 'White Bread',
        },
      },
    },
  },
  foodId: 'mendosa-white-bread',
  sourceName: 'mendosa',
}

export default testData
