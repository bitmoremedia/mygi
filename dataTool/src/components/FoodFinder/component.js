import React, { Component } from 'react';
import _values from 'lodash/values';

import { search } from '../../utils/search';

import Divider from '../common/Divider';
import { Grid, Row, Col } from '../common/Grid';
import Button from '../common/Button';

import AddEditFood from '../AddEditFood';

import { Container, Heading, SubHeading } from './styles';
import FoodList from './FoodList';
import DataSourceFoodList from './DataSourceFoodList';

class FoodFinder extends Component {

  constructor(props, context) {
    super(props, context);
    this.state = {
      showAddFood: false,
    };
    this.toggleAddFood = this.toggleAddFood.bind(this);
  }

  componentWillReceiveProps() {
    this.setState({
      showAddFood: false
    });
  }

  toggleAddFood() {
    this.setState({
      showAddFood: !this.state.showAddFood
    });
  }

  render(){
    const { toggleAddFood } = this;
    const { showAddFood } = this.state;
    const { foodList, dataSources, foodId, sourceName } = this.props;

    const thisFood = dataSources[sourceName].data[foodId];
    const allDataSourcesArray = _values(dataSources);

    const filteredFoodList = search({
      searchText: thisFood.name,
      searchList: _values(foodList),
      searchField: 'name'
    });

    let allDataSources = [];
    allDataSourcesArray.forEach((source)=>{
      if (source.name !== sourceName){
        let sourceData = _values(source.data);
        // add source name as data attribute of each food item
        sourceData = sourceData.map(food=> { return {...food, source: source.name}});
        allDataSources = allDataSources.concat(sourceData);
      }
    });

    const filteredDataSources = search({
      searchText: thisFood.name,
      searchList: allDataSources,
      searchField: 'name'
    });

    const renderToggleAddFoodButton = () => {
      const buttonText = (this.state.showAddFood) ? 'Cancel' : 'Add';
      return <Button onClick={()=>toggleAddFood()}>{buttonText}</Button>;
    }

    return (
      <Container>
        <Heading>
          {thisFood.name} [{thisFood.gi}]
        </Heading>
        <Divider />
          <Grid>
            <Row>
              <Col xs={12} md={4}>
                <SubHeading>
                  Food List {renderToggleAddFoodButton()}
                </SubHeading>
                {showAddFood && <AddEditFood />}
                <FoodList list={filteredFoodList} sourceName={sourceName} sourceId={foodId} />
              </Col>
              <Col xs={12} md={7}>
                <SubHeading>
                  Other Data Sources
                </SubHeading>
                <DataSourceFoodList list={filteredDataSources} />
              </Col>
            </Row>
          </Grid>
      </Container>
    );
  }
};

export default FoodFinder;
