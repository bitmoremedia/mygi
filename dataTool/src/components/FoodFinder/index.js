import React, { Component } from 'react';
import _values from 'lodash/values';

import { search } from '../../utils/search';

import Divider from '../common/Divider';
import { Grid, Row, Col } from '../common/Grid';

import { Container, Heading, FoodHeading, SubHeading } from './styles';
import FoodList from './FoodList';
import DataSourceFoodList from './DataSourceFoodList';

class FoodFinder extends Component {

  constructor(props, context) {
    super(props, context);
    this.state = {
      showHelp: false
    };
  }

  render(){
    const { foodList, dataSources, foodId, sourceName } = this.props;
    const thisFood = dataSources[sourceName].data[foodId];
    const allDataSourcesArray = _values(dataSources);

    const filteredFoodList = search({
      searchText: thisFood.name,
      searchList: _values(foodList),
      searchField: 'name'
    });

    let allDataSources = [];
    allDataSourcesArray.forEach((source)=>{
      if (source.name !== sourceName){
        let sourceData = _values(source.data);
        // add source name as data attribute of each food item
        sourceData = sourceData.map(food=> { return {...food, source: source.name}});
        allDataSources = allDataSources.concat(sourceData);
      }
    });

    const filteredDataSources = search({
      searchText: thisFood.name,
      searchList: allDataSources,
      searchField: 'name'
    });

    return (
      <Container>
        <Heading>Find
          <FoodHeading>
            {thisFood.name} [{thisFood.gi}]
          </FoodHeading>
        </Heading>
        <Divider />
          <Grid>
            <Row>
              <Col xs={12} md={4}>
                <SubHeading>
                  Food List
                </SubHeading>
                <FoodList list={filteredFoodList} />
              </Col>
              <Col xs={12} md={7}>
                <SubHeading>
                  Other Data Sources
                </SubHeading>
                <DataSourceFoodList list={filteredDataSources} />
              </Col>
            </Row>
          </Grid>
      </Container>
    );
  }
};

export default FoodFinder;
