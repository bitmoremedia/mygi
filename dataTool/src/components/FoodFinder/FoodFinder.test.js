import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import FoodFinder from './component'

import testData from '../../utils/testData'

test('component renders without error', () => {
  const wrapper = shallow(
    <FoodFinder
      foodList={testData.foodList}
      dataSources={testData.dataSources}
      foodId={testData.foodId}
      sourceName={testData.sourceName}
    />,
  )
  expect(wrapper.exists()).toBe(true)
})

// SNAPSHOT TESTS

test('component renders as expected', () => {
  const wrapper = shallow(
    <FoodFinder
      foodList={testData.foodList}
      dataSources={testData.dataSources}
      foodId={testData.foodId}
      sourceName={testData.sourceName}
    />,
  )
  expect(toJson(wrapper)).toMatchSnapshot()
})
