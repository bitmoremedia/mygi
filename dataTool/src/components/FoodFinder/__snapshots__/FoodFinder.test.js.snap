exports[`test component renders as expected 1`] = `
<styled.div>
  <styled.h1>
    White Bread
     [
    77
    ]
  </styled.h1>
  <Component />
  <Row>
    <Col
      md={4}
      xs={12}>
      <styled.h4>
        Food List 
        <Component
          danger={false}
          onClick={[Function]}>
          Add
        </Component>
      </styled.h4>
      <Connect(Component)
        list={
          Array [
            Object {
              "category": "Bread",
              "gi": 68,
              "id": "efgh-456",
              "name": "White Bread",
              "sources": Object {
                "havard": "havard-white-bread",
              },
            },
          ]
        }
        sourceId="mendosa-white-bread"
        sourceName="mendosa" />
    </Col>
    <Col
      md={7}
      xs={12}>
      <styled.h4>
        Other Data Sources
      </styled.h4>
      <Component
        list={
          Array [
            Object {
              "gi": 65,
              "id": "havard-white-bread",
              "name": "White Bread",
              "source": "havard",
            },
          ]
        } />
    </Col>
  </Row>
</styled.div>
`;
